common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -739,31 +739,52 @@ def definition_to_xml(self, resource_fs):
             ele.set('src', self.handout)
             xml.append(ele)
 
+        transcripts = {}
         if self.transcripts is not None:
-            # sorting for easy testing of resulting xml
-            for transcript_language in sorted(self.transcripts.keys()):
-                ele = etree.Element('transcript')
-                ele.set('language', transcript_language)
-                ele.set('src', self.transcripts[transcript_language])
-                xml.append(ele)
+            transcripts.update(self.transcripts)
 
         edx_video_id = clean_video_id(self.edx_video_id)
         if edxval_api and edx_video_id:
             try:
                 # Create static dir if not created earlier.
                 resource_fs.makedirs(EXPORT_IMPORT_STATIC_DIR, recreate=True)
 
-                xml.append(
-                    edxval_api.export_to_xml(
-                        video_id=edx_video_id,
-                        resource_fs=resource_fs,
-                        static_dir=EXPORT_IMPORT_STATIC_DIR,
-                        course_id=unicode(self.runtime.course_id.for_branch(None))
-                    )
+                # Backward compatible exports
+                # edxval exports new transcripts into the course OLX and returns a transcript
+                # files map so that it can also be rewritten in old transcript metadata fields
+                # (i.e. `self.transcripts`) on import and older open-releases (<= ginkgo),
+                # who do not have deprecated contentstore yet, can also import and use new-style
+                # transcripts into the their openedX instances.
+
+                exported_metadata = edxval_api.export_to_xml(
+                    video_id=edx_video_id,
+                    resource_fs=resource_fs,
+                    static_dir=EXPORT_IMPORT_STATIC_DIR,
+                    course_id=unicode(self.runtime.course_id.for_branch(None))
                 )
+                # Update xml with edxval metadata
+                xml.append(exported_metadata['xml'])
+
+                # we don't need sub if english transcript
+                # is also in new transcripts.
+                new_transcripts = exported_metadata['transcripts']
+                transcripts.update(new_transcripts)
+                if new_transcripts.get('en'):
+                    xml.set('sub', '')
+
+                # Update `transcripts` attribute in the xml
+                xml.set('transcripts', json.dumps(transcripts))
```

Comment:
@nasthagiri I needed to make changes in `xmodule/xml_module.py` to keep `XmlParserMixin` from overriding/cleaning the updated attributes in video xml. 

Let me know if there is any concern in this regard, or if there are better ways to do it with/without touching `xmodule/xml_module.py`.

Reply:
@ormsbee do you have any better ideas for the team other than what they have here?

Reply:
It took me a little while to grok, but I think I get it now. It'd be cleaner to break backwards compatibility, but that would be problematic for folks who are running courses across multiple versions (e.g. the edX partners still running ginkgo), and I can't think of a better way to keep that compatibility.

Video module serialization should be moved entirely to XBlock mechanisms, but that's a much bigger task to take on. Given that, I think the small amount of mucking in `xml_module.py` is unfortunate but it's the practical thing to do. :(